proposals/lip-0043.md

```diff
@@ -385,52 +486,123 @@ Let `trs` be a transaction with module ID `MODULE_ID_INTEROPERABILITY` and comma
+    ownChainAccount = {
+        "name": trs.params.ownName,
+        "chainID": trs.params.ownChainID,
+        "nonce": 0
+    }
+    create an entry in the own chain data substore with
+        storeKey = EMPTY_BYTES
+        storeValue = encode(ownChainAccountSchema, ownChainAccount)
+
+    # Emit chain account updated event.
+    emitEvent(
+        module = MODULE_NAME_INTEROPERABILITY,
+        name = EVENT_NAME_CHAIN_ACCOUNT_UPDATED,
+        data = mainchainAccount,
+        topics = [CHAIN_ID_MAINCHAIN]
+    )
+
+    # Send registration CCM to the mainchain.
+    # Notice that we do not use the send function because the channel
+    # has not been activated yet.
     registrationCCMParams = {
-        "networkID": MAINCHAIN_NETWORK_ID,
-        "name": MAINCHAIN_NAME,
+        "chainID": CHAIN_ID_MAINCHAIN,
+        "name": CHAIN_NAME_MAINCHAIN,
         "messageFeeTokenID": mainchainChannel.messageFeeTokenID
     }
-    ```
-
-* An entry in the chain validators substore as specified in [LIP 0045][lip-0045#validators-substore] is created as:
-  * `storePrefix`: `STORE_PREFIX_CHAIN_VALIDATORS`
-  * `storeKey`: `uint32be(MAINCHAIN_ID)`.
-  * `storeValue`: `mainchainValidators` where `mainchainValidators` is the validators object, serialized according to the `validatorsSchema` schema as defined in [LIP 0045][lip-0045], and initialized as:
-    * The `mainchainValidators.activeValidators` property is equal to the `trs.params.mainchainValidators` property.
-    * The `mainchainValidators.certificateThreshold` property is equal to the `THRESHOLD_MAINCHAIN`.
-* An entry in the outbox root substore as specified in [LIP 0045][lip-0045] is created as:
-  * `storePrefix`: `STORE_PREFIX_OUTBOX_ROOT`.
-  * `storeKey`:  `uint32be(MAINCHAIN_ID)`.
-  * `storeValue`: The property `mainchainChannel.outbox.root` serialized according to the `outboxRootSchema` schema defined in Introduce Interoperability module LIP.
-* An entry in the own chain account data substore as specified in [LIP 0045][lip-0045] is created as:
-  * `storePrefix` : `STORE_PREFIX_CHAIN_DATA`
-  * `storeKey`: `uint32be(0)`.
-  * `storeValue`: `ownChainAccount` where `ownChainAccount` is the own chain account object, serialized according to the `ownChainAccountSchema` schema defined in Introduce Interoperability module LIP, and initialized as:
-    * `ownChainAccount.ID` = `trs.params.ownChainID`.
-    * `ownChainAccount.name` = `trs.params.ownName`.
-    * `ownChainAccount.nonce` = 0.
+
+    ccm = {
+        "nonce": ownChainAccount.nonce,
+        "module": MODULE_NAME_INTEROPERABILITY,
+        "crossChainCommand": CROSS_CHAIN_COMMAND_REGISTRATION,
+        "sendingChainID": ownChainAccount.chainID,
+        "receivingChainID": CHAIN_ID_MAINCHAIN,
+        "fee": 0,
+        "status": CCM_OK,
+        "params": encode(registrationCCMParamsSchema, registrationCCMParams) # registrationCCMParamsSchema is defined in LIP0049
+    }
+
+    addToOutbox(CHAIN_ID_MAINCHAIN, ccm)
```

Comment:
I think here it would be more consistent to use the `send` function so that `addToOutbox` is only used in the transport layer functions send / error / forward / bounce.

Reply:
same as above